cmd/snap-confine/mount-support.c

```diff
@@ -817,6 +817,16 @@ void sc_setup_user_mounts(struct sc_apparmor *apparmor, int snap_update_ns_fd,
 					 "XDG_RUNTIME_DIR=%s", xdg_runtime_dir);
 			envp[last_env++] = xdg_runtime_dir_env;
 		}
+		const char *snap_instance_user_data =
+		    getenv("SNAP_INSTANCE_USER_DATA");
+		char snap_instance_user_data_env[PATH_MAX];
+		if (snap_instance_user_data != NULL) {
+			sc_must_snprintf(snap_instance_user_data_env,
+					 sizeof(snap_instance_user_data_env),
+					 "SNAP_INSTANCE_USER_DATA=%s",
+					 snap_instance_user_data);
+			envp[last_env++] = snap_instance_user_data_env;
```

Comment:
envp is just three entries long and must terminate with an empty entry. There are three variables that we possibly add: `SNAPD_DEBUG`, `XDG_RUNTIME_DIR` and `SNAP_INSTANCE_USER_DATA`. This means we violate the spec of `fexecve` when all of those are used at the same time. Please enlarge `envp` to `4`.